src/main/java/com/ot/restaurant/ioc/DependencyContainer.java

```diff
@@ -0,0 +1,54 @@
+package com.ot.restaurant.ioc;
+
+import java.io.File;
+
+public class DependencyContainer<T> {
```

Comment:
Probably you just need to rename this class to `IoCContainerInitializer` and then here you can just have somethin like this:
```
public static void initialize() {
     initializeRepositories();
     initializeControllers();
}
```